tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -23,75 +23,768 @@ THE SOFTWARE.
+__global__ void hipLaunchKernelStructFunc14(
+                    hipLaunchKernelStruct_t14 hipLaunchKernelStruct_,
+                    bool* result_d14) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    hipLaunchKernelStruct_.writeint[x] = 1;
+    // set the result to true if the condition met
+    result_d14[x] =  ((hipLaunchKernelStruct_.readint == 1)
+                      && (hipLaunchKernelStruct_.writeint[x] == 1));
+}
+
+// Passing struct with struct with dynamic memory, new int
+// the heap memory will be accessed from device
+__global__ void hipLaunchKernelStructFunc15(
+                    hipLaunchKernelStruct_t15 hipLaunchKernelStruct_,
+                    bool* result_d15) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    result_d15[x] =  ((hipLaunchKernelStruct_.c1 == 'c')
+                       && (hipLaunchKernelStruct_.heapmem[x] == 1));
+}
+
+// Passing simple template struct
+__global__ void hipLaunchKernelStructFunc16(
+                    hipLaunchKernelStruct_t16<char> hipLaunchKernelStruct_,
+                    bool* result_d16) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    result_d16[x] =  (hipLaunchKernelStruct_.t1 == 'c');
+}
+
+// Passing simple explicit template struct
+__global__ void hipLaunchKernelStructFunc17(
+                    hipLaunchKernelStruct_t17<int> hipLaunchKernelStruct_,
+                    bool* result_d17) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    result_d17[x] =  (hipLaunchKernelStruct_.t1 == 1);
+}
+
+// Passing struct and write to struct memory using __device__ func()
+__global__ void hipLaunchKernelStructFunc18(
+                    hipLaunchKernelStruct_t18 hipLaunchKernelStruct_,
+                    bool* result_d18) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    hipLaunchKernelStruct_.setChar('c');
+    // set the result to true if the condition met
+    result_d18[x] =  (hipLaunchKernelStruct_.getChar() == 'c');
+}
+
+// Passing simple user defined stack implemenration,  using __device__ func()
+__global__ void hipLaunchKernelStructFunc19(
+                    hipLaunchKernelStruct_t19 hipLaunchKernelStruct_) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // stack should be empty after the kernel execustion, verify on host side
+    hipLaunchKernelStruct_.popMe();
+}
+
+// Passing out of order initalized struct, access in-order
+__global__ void hipLaunchKernelStructFunc20(
+                    hipLaunchKernelStruct_t20 hipLaunchKernelStruct_,
+                    bool* result_d20) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // accessing struct members in order
+    result_d20[x] = (hipLaunchKernelStruct_.name = 'A'
+    // strcmp(hipLaunchKernelStruct_.name, "AMD") -> strcmp is not broken
```

Comment:
I am not quite get it. name is a member of char type, why is it compared with a string?

Reply:
initially i tried with a char* name = "AMD" and called strcmp(). As it was not working i changed to a char instead a pointer

Reply:
strcmp is not a device function, therefore cannot be called by a device function.